src/httpserver.cpp

```diff
@@ -162,12 +162,10 @@ static bool ClientAllowed(const CNetAddr& netaddr)
 static bool InitHTTPAllowList()
 {
     rpc_allow_subnets.clear();
-    CNetAddr localv4;
-    CNetAddr localv6;
-    LookupHost("127.0.0.1", localv4, false);
-    LookupHost("::1", localv6, false);
-    rpc_allow_subnets.push_back(CSubNet(localv4, 8));      // always allow IPv4 local subnet
-    rpc_allow_subnets.push_back(CSubNet(localv6));         // always allow IPv6 localhost
+    const std::optional<CNetAddr> localv4{LookupHost("127.0.0.1", false)};
+    const std::optional<CNetAddr> localv6{LookupHost("::1", false)};
+    rpc_allow_subnets.push_back(CSubNet(localv4.value(), 8));      // always allow IPv4 local subnet
+    rpc_allow_subnets.push_back(CSubNet(localv6.value()));         // always allow IPv6 localhost
```

Comment:
perhaps no need for intermediary variables?
```suggestion
    rpc_allow_subnets.push_back(CSubNet{LookupHost("127.0.0.1", false).value(), 8});      // always allow IPv4 local subnet
    rpc_allow_subnets.push_back(CSubNet{LookupHost("::1", false).value()});               // always allow IPv6 localhost
```